misk/src/main/kotlin/misk/client/ClientInvocationHandler.kt

```diff
@@ -56,8 +59,12 @@ internal class ClientInvocationHandler(
           clientBuilder.build()
         }
 
-    methodName to retrofit.newBuilder()
+    val retrofitBuilder = retrofit.newBuilder()
         .client(actionSpecificClient)
+
+    if (tracer != null) retrofitBuilder.callFactory(TracingCallFactory(actionSpecificClient, tracer))
```

Comment:
Please add this to the retrofit object created in the `TypedHttpClientModule` rather than doing it here. All global retrofit properties should be established in the module; anything added here is specific to an individual action

Reply:
So, I originally had it in the `TypedHttpClientModule`, but that broke once the `ClientInvocationHandler` changes landed. The issue was that the `TracingCallFactory` wasn't  being given the correct `OkHttpClient` after the changes. The `OkHttpClient` in `TypedHttpClientModule` isn't the client performing the actual calls, rather a template to generate the `actionSpecificClient`. Therefore, to get tracing to work properly, it needs to be installed in the `ClientInvocationHandler` the the call factory can be applied to the correct `OkHttpClient`